docs/development/gittuf-recovery.md

```diff
@@ -0,0 +1,155 @@
+  Number: 463
+
+entry cf2f7e1ad70ef5b1be92bf0e8bd9e4461e36571c
+
+  Ref:    refs/heads/update-roadmap
+  Target: 00bb3032e50354e04f567cad157ccb9137d42512
+  Number: 462
+
+entry 235312e7f06c75d4be0ffb3ff87388f10be9882f
+
+  Ref:    refs/heads/main
+  Target: c4c661618dc96351792ef542fe3cd29b15323d94
+  Number: 461
+
+entry e900a5845e92a8a764e4f10eeaeee2a7a00a53c2
+
+  Ref:    refs/gittuf/attestations
+  Target: a1028ccedff72017efc47d366a0be5c300216b65
+  Number: 460
+
+entry db05a0f1dcb6121e6aceb130d3ca9da1e67cd690
+
+  Ref:    refs/gittuf/attestations
+  Target: db598f89434648b12b290f628c4db0f8e1f5156e
+  Number: 459
+
+entry cb639d702af5d22246039b3978d58bd1af632bb0
+
+  Ref:    refs/heads/main
+  Target: 68b94ea3f289f335c964b0d286df7a00404b4af9
+  Number: 458
+```
+
+In our case, the entry is numbered 461. To restore the repository to a
+known-good state, we must first create a _skip annotation_ that informs gittuf
+that the entry should not be processed, and there is another entry that will
+restore the repository to a known good state.
+
+To create this skip annotation, we run the `gittuf rsl annotate` command, like
+so:
+
+```bash
+gittuf rsl annotate --skip --local-only --message "GitHub API timing issues" 235312e7f06c75d4be0ffb3ff87388f10be9882f
+```
+
+After performing this action, running `gittuf rsl log` will now show the entry
+as skipped:
+
+```bash
+entry 235312e7f06c75d4be0ffb3ff87388f10be9882f (skipped)
+
+  Ref:    refs/heads/main
+  Target: c4c661618dc96351792ef542fe3cd29b15323d94
+  Number: 461
+
+    Annotation ID: 194200eb770c8025170dd9bcf2dc89f760da8017
+    Skip:          yes
+    Number:        474
+    Message:
+      GitHub API timing issues
+```
+
+## 2. Recording the Last Known-Good State
+
+Once the invalid entry has been revoked, the second step is to record in the RSL
+an entry for the _last known-good state_ for the reference in question. This is
+normally the commit that preceded the invalid one(s). In our case, the last
+commit to the `main` branch has a hash of
+`68b94ea3f289f335c964b0d286df7a00404b4af9`:
+
+```bash
+entry cb639d702af5d22246039b3978d58bd1af632bb0
+
+  Ref:    refs/heads/main
+  Target: 68b94ea3f289f335c964b0d286df7a00404b4af9
+  Number: 458
+```
+
+To record an RSL entry for this entry again, you should rewind the branch in
+question until the latest commit on the branch is the one preceding any invalid
+ones. In our case, we must rewind the `main` branch by one commit. To do this,
+you may use the `update-ref` command from Git:
+
+```bash
+git update-ref main main~1 # Where the number after the ~ is the number of commits to rewind by.
+```
+
+Next, you must record the RSL entry with `gittuf rsl record`:
+
+```bash
+gittuf rsl record --local-only main
+```
+
+Finally, pull the invalid changes back onto the main branch with `git pull`.
+
+## 3. Recording the Latest State
+
+Once the previous-good commit has been recorded again in the RSL, you must
```

Comment:
The changes are already in `main`. This is about recording the "actually good" changes back in the RSL. It's important to note that this subsequent RSL entry needs to be signed by someone who can meet the policy threshold. Since this is a document for gittuf/gittuf only, maybe some of those details should be added in?